src/fsharp/FSComp.txt

```diff
@@ -17,6 +17,7 @@ undefinedNameTypeParameter,"The type parameter %s is not defined."
 undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined."
 replaceWithSuggestion,"Replace with '%s'"
 missingElseBranch,"The 'if' expression is missing an 'else' branch. The 'then' branch has type '%s'. Because 'if' is an expression, and not a statement, add an 'else' branch which returns a value of the same type."
+ifExpression,"The 'if' expression needs to return type '%s' in order to satisfy context type requirements. It currently returns type '%s'."
```

Comment:
'Returns' may be a bit of a misnomer here - the expression itself has a type. Maybe just: "The 'if' expressions needs to have type '%s' in order to satisfy context type requirements. It currently has type '%s'."

Also out of interest what is a 'context type requirement'?